lib/active_model_serializers/adapter/json_api/relationship.rb

```diff
@@ -34,6 +34,11 @@ def as_json
 
         private
 
+        def ensure_valid!(json)
```

Comment:
> Why does it need to be a separate endpoint? What if it's an existing one with a parameter?

Then you'd be linking to a collection of posts, instead of a collection of comments, which is invalid (even if the comments are sideloaded via an ?include parameter).

> Why not just make an an endpoint for fetching that data?

Because I'd rather not code and test a completely unused endpoint just because `relationships: { comments: {} }` is not valid jsonapi.

> What about a relationships endpoint for that resource?

Not needed, this relationship is only fetched by my web application when sideloading.

> The answers to those questions are better determined by the application and API design.

Agree. With the above answers, I am saying it is a reasonable API design to not have a separate endpoint, which means no `links` can be present.